src/Mod/Arch/archmake/make_reinforcement_linear.py

```diff
@@ -0,0 +1,180 @@
+# thus 2 of these 3 are needed, the third is calculated
+# Amount, Distance, Spacing
+#
+# If two are given
+# calculate the third
+#
+# If all are given
+# ATM: print, not yet supported
+# do some test need to be make if there is some rest
+# What if there is a rest.
+# rest:
+# Distance is fixed, fix Amount, change Spacing
+# Amount is fixed, fix Distance, change Spacing
+# Spacing is fixed, fix Amount, change Distance
+# in any case print a message what was changed
+#
+# If only one is given:
+# ATM: print, not yet supported
+#
+# If none of them is given:
+# ATM: print, not yet supported
+#
+
+
+def makeReinforcementLinear(
+    base_rebar,
+    amount=None,
+    spacing=None,
+    distance=None,
+    direction=vec(0, 0, 1),
+    base_placement=FreeCAD.Placement(),
+    name="ReinforcementLinear"
+):
+    """
+    makeReinforcementLinear(
+        base_rebar,
+        [amount],
+        [spacing],
+        [distance],
+        [direction],
+        [base_placement],
+        [name]
+    )
+    Adds a linear reinforcement object.
+    """
+
+    if not FreeCAD.ActiveDocument:
+        FreeCAD.Console.PrintError("No active document. Aborting\n")
+        return
+
+    # print("amount: {}".format(amount))
+    # print("distance: {}".format(distance))
+    # print("spacing: {}".format(spacing))
+    if (
+        (amount is None and spacing is None)
+        or (amount is None and distance is None)
+        or (distance is None and spacing is None)
+        or (amount is None and distance is None and spacing is None)
+        or (
+            amount is not None
+            and distance is not None
+            and spacing is not None
+        )
+    ):
+        FreeCAD.Console.PrintError(
+            "This combination of parameter is not yet supported. Aborting\n"
+        )
+        return
+
+    obj = FreeCAD.ActiveDocument.addObject(
+        "Part::FeaturePython",
+        "ReinforcementLinear"
+    )
+    obj.Label = translate("Arch", name)
+
+    from archobjects.reinforcement_linear import ReinforcementLinear
+    ReinforcementLinear(obj)
+    if FreeCAD.GuiUp:
+        import archviewproviders.view_reinforcement_linear as view_linear
+        view_linear.ViewProviderReinforcementLinear(obj.ViewObject)
+
+    obj.BaseRebar = base_rebar
+    obj.BasePlacement = base_placement
+    obj.Direction = direction
+    obj.Document.recompute()
+
+    if distance is None:
+        obj.FixedAttribut = "Amount"
+        obj.Amount = amount
+        obj.Spacing = spacing
+
+    if amount is None:
+        obj.FixedAttribut = "Distance"
+        obj.Distance = distance
+        obj.Spacing = spacing
+
+    if spacing is None:
+        obj.FixedAttribut = "Spacing"
+        obj.Distance = distance
+        obj.Amount = amount
+
+    obj.Document.recompute()
+
+    # mark base_rebar obj for recompute to make it collect its new child
+    base_rebar.touch()
+    obj.Document.recompute()
+    return obj
+
+
+"""
+# test above code
```

Comment:
I suggest providing a test script. This is what I did in Draft, inside `drafttests.draft_test_objects`. I want to do the same with Arch, but I haven't had time. We need to remake all unit tests in general.

Reply:
BTW: in this Draft unit test you do use camelCase make method names ???

Reply:
At the moment yes, because that is what existed before. But now we have snake case methods, so we will change the unit tests. I will work on that in the next few days.

Reply:
ahh ok got it.

Reply:
ATM I copy test code. I would wait until we have some better test frame work in Arch like we have in Draft.